src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,319 @@
+from collections import (
+    defaultdict,
+)
+import json
+import logging
+import os
+from typing import (
+    ClassVar,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Union,
+)
+from uuid import (
+    UUID,
+    uuid5,
+)
+
+import attr
+from furl import (
+    furl,
+)
+from github import (
+    Github,
+)
+from github.Repository import (
+    Repository,
+)
+
+from humancellatlas.data.metadata.api import (
+    JSON,
+)
+from humancellatlas.data.metadata.helpers.exception import (
+    reject,
+    require,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(frozen=True, kw_only=True, auto_attribs=True)
+class JsonFile:
+    """
+    A JSON file in the staging area.
+    """
+    uuid: str
+    version: str
+    name: str
+    content: JSON
+
+    def __attrs_post_init__(self):
+        pass
```

Comment:
https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#117disabling-sections-of-code

otherwise it looks weird when reviewing all changes including the drop commit

```suggestion
    validator: ClassVar[SchemaValidator] = SchemaValidator()
    
    def __attrs_post_init__(self):
        self.validator.validate_json(…)
```